### PR TITLE
test: basic tests for CLI covering SBOM and advisories

### PR DESCRIPTION
**Pull request by helio-frota**
using fake and basic data and better than nothing for coverage

## Summary by Sourcery

Add integration-style tests for the CLI SBOM and advisory commands using mocked HTTP APIs.

Build:
- Add wiremock and async process-capable tokio as CLI dev-dependencies for test support.

Tests:
- Add CLI tests covering SBOM listing, querying, pagination, sorting, retrieval, deletion, duplicate handling, and pruning workflows.
- Add CLI tests covering advisory listing, querying, pagination, and pruning scenarios.

## Summary by Sourcery

Add integration-style CLI tests for SBOM and advisory commands using a mocked HTTP API.

Build:
- Add wiremock and process-capable tokio as CLI dev-dependencies to support async CLI testing.

Tests:
- Add CLI tests for SBOM listing with multiple output formats, filtering, pagination, sorting, retrieval, deletion, duplicate detection, and pruning workflows.
- Add CLI tests for advisory listing with filtering and pagination, and pruning scenarios including dry-run, age-based, and keep-latest options.

### CODE REVIEW

File: cli/tests/cli.rs [hunk 1]
```diff
@@ -0,0 +1,841 @@
...
+        .mount(&server)
+        .await;
+
+    let output = Command::new(env!("CARGO_BIN_EXE_trustify"))
+        .args(["--url", &server.uri(), "sbom", "list"])
+        .output()
+        .await
+        .unwrap();
+
+    assert!(output.status.success());
```
[the remark this answers is not in the document]
Comment (helio-frota) [reply cli/tests/cli.rs:66]: I removed this test... these are basic tests only

[the remark this answers is not in the document]
Comment (helio-frota) [reply cli/tests/cli.rs:66]: test removed... I'm not fully in the details just writing some sort of skeleton test with fake mock data

**Comment by helio-frota** [comment]
@sourcery-ai review

**Comment by helio-frota** [comment]
> The CLI tests contain a lot of repeated patterns for starting a MockServer,

wiremock itself does this https://github.com/LukeMathWalker/wiremock-rs/blob/main/tests/request_header_matching.rs a lot of times `  let mock_server = MockServer::start().await;`

> You are declaring tokio both in [dependencies] and [dev-dependencies] with different feature sets; it may be cleaner and less error‑prone to keep a single tokio declaration (e.g. in the workspace or main [dependencies]) that includes all needed features.

defensive approach..nothing wrong for me as the feature addition was for dev-dep only

> Several tests create temporary JSON files (e.g. duplicates/prune outputs) using fixed filenames in temp_dir

they are discarded later

```
running 24 tests
test cli_sbom_duplicates_find_default ... ok
test cli_advisory_prune_older_than ... ok
test cli_sbom_delete_by_id_dry_run ... ok
test cli_advisory_prune_keep_latest ... ok
test cli_advisory_list ... ok
test cli_sbom_delete_by_query_dry_run ... ok
test cli_sbom_delete_by_id ... ok
test cli_advisory_list_with_query ... ok
test cli_advisory_prune_dry_run ... ok
test cli_sbom_duplicates_delete_dry_run ... ok
test cli_sbom_duplicates_find_custom_params ... ok
test cli_advisory_list_with_pagination ... ok
test cli_sbom_list_name_format ... ok
test cli_sbom_list_full_format ... ok
test cli_sbom_get_by_id ... ok
test cli_sbom_list_short_format ... ok
test cli_sbom_prune_by_label ... ok
test cli_sbom_list_with_sort ... ok
test cli_sbom_list_with_query ... ok
test cli_sbom_prune_dry_run ... ok
test cli_sbom_list_id_format ... ok
test cli_sbom_list_with_pagination ... ok
test cli_sbom_prune_published_before ... ok
test cli_sbom_prune_keep_latest ... ok

test result: ok. 24 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.36s

➜  cli git:(cli-mock-tests) git status
On branch cli-mock-tests
nothing to commit, working tree clean
```

@sourcery-ai :-1:

**Review by i386x — APPROVED** [review]
LGTM. I like this. Nice work.

**Comment by helio-frota** [comment]
@i386x thanks, this helped me to understand a bit more what the CLI is doing... not 100% of course but helped a bit

unfortunatelly the tests are really basic but it covers some execution paths of the CLI and may help for future additions/deletions (probably)
